### ISSUE TITLE
Effect of  `requires_grad=False` in init_hidden of LSTM/GRU encoder/decoder

### ISSUE THREAD

**Issue by Nokimann** [ibm-research-tokyo/graph_grammar#3]
Example the code in graph_grammar/src/graph_grammar/nn/decoder/GRUDecoder

```
    def init_hidden(self):
        self.hidden_dict['h'] = torch.zeros((self.num_layers,
                                             self.batch_size,
                                             self.hidden_dim),
                                            requires_grad=False)
        if self.use_gpu:
            self.hidden_dict['h'] = self.hidden_dict['h'].cuda()
```

At first glance, it looks like no gradient flow.
But, when we apply `feed_hidden` during `decode` function in `GrammarSeq2SeqVAE`,
immediately `requires_grad` of `hidden_dict['h']` recovers to True.

Does this affect when training a model?

I'm customizing your code...

**Comment by kanojikajino** [comment]
I'm sorry but I don't remember the exact reason why I set `requires_grad=False`; it should be `requires_grad=True` so that the error can be backpropagated through the hidden states.
